proxy/src/coco/source.rs

```diff
@@ -42,8 +42,27 @@ pub struct Person {
     pub avatar: String,
 }
 
-/// Representation of a code commit.
+/// Commit statistics.
+#[derive(Serialize)]
+pub struct CommitStats {
+    /// Additions.
+    pub additions: u64,
+    /// Deletions.
+    pub deletions: u64,
+}
+
+/// Representation of a changeset between two revs.
 pub struct Commit {
+    /// The commit header.
+    pub header: CommitHeader,
+    /// The change statistics for this commit.
+    pub stats: CommitStats,
+    /// The changeset introduced by this commit.
+    pub diff: surf::diff::Diff,
+}
+
+/// Representation of a code commit.
+pub struct CommitHeader {
```

Comment:
I'm not sure the renaming makes sense to me here. I think the `Commit` struct above that contains the `Diff` could be named `Changeset` instead. A `Commit` is the details of some change made, whereas the above is capturing the differences between two of those changes.

Reply:
Yeah, that's actually what I had previously, but then realized I would have to change the API endpoints to match :cry:, because it's currently `/commit/<sha>`. I was also thinking that within the scope of the UI, calling it a commit is fine. GitHub does the same, eg. https://github.com/radicle-dev/radicle-upstream/commit/8bf7ea37d10b9e9c8352abc42e3e3d4d9398dd19, even though this isn't technically "just" the commit.

Reply:
But what do _we_ want?! If it makes sense to change the endpoints now I'd say we go for it.

Reply:
I think from the UI perspective, a commit is everything that is shown on the commit page -- ie. it doesn't have to mean the same thing as from surf's perspective.

Reply:
Cool, that makes sense.